allennlp/tests/data/dataset_readers/textcat_test.py

```diff
@@ -0,0 +1,46 @@
+# pylint: disable=no-self-use,invalid-name
+import pytest
+
+from allennlp.data.dataset_readers import TextCatReader
+from allennlp.common.util import ensure_list
+from allennlp.common.testing import AllenNlpTestCase
+
+
+class TestTextCatReader():
+
+    @pytest.mark.parametrize("lazy", (True, False))
+    def test_read_from_file_ag(self, lazy):
+        reader = TextCatReader(lazy=lazy)
+        ag_path = AllenNlpTestCase.FIXTURES_ROOT / "data" / "textcat" / "ag.jsonl"
+        instances = reader.read(ag_path)
+        instances = ensure_list(instances)
+
+        instance1 = {"tokens": ['Memphis', 'Rout', 'Still', 'Stings', 'for', 'No', '.', '14',
+                                'Louisville', ';', 'Coach', 'Petrino', 'Vows', 'to', 'Have',
+                                'Team', 'Better', 'Prepared', '.', 'NASHVILLE', ',', 'Tenn.',
+                                'Nov', '3', ',', '2004', '-', 'Louisville', '#', '39;s', '30-point',
+                                'loss', 'at', 'home', 'to', 'Memphis', 'last', 'season', 'is', 'still',
+                                'a', 'painful', 'memory', 'for', 'the', 'Cardinals', '.'],
+                     "label": "2"}
```

Comment:
I'm guessing you didn't pick this format, but having numbers as labels does funny things to your outputs, unless you pass `skip_indexing` to the label field.  You might consider having that as an option for the dataset reader, so you don't re-number these numbers.